invenio_rdm_records/collections/api.py

```diff
@@ -91,8 +91,21 @@ def resolve(cls, *, id_=None, slug=None, ctree_id=None, depth=2):
     @classmethod
     def resolve_many(cls, ids_=None, depth=2):
         """Resolve many collections by ID."""
-        _ids = ids_ or []
-        return [cls(c, depth) for c in cls.model_cls.read_many(_ids)]
+        if not ids_:
+            return []
```

Comment:
Very minor comment: why checking this year, and not delegating it to the caller?
The caller could avoid calling the method if no `ids` given.
Otherwise, it feels like that we should check the values of all params, including `depth`.